pydantic/decorator.py

```diff
@@ -246,7 +249,8 @@ def check_positional_only(cls, v: Optional[List[str]]) -> None:
                 keys = ', '.join(map(repr, v))
                 raise TypeError(f'positional-only argument{plural} passed as keyword argument{plural}: {keys}')
 
-            @validator(V_DUPLICATE_KWARGS, check_fields=False, allow_reuse=True)
+            @field_validator(V_DUPLICATE_KWARGS, check_fields=False, allow_reuse=True)
+            @classmethod
             def check_duplicate_kwargs(cls, v: Optional[List[str]]) -> None:
```

Comment:
Do you know what's going on here? No tests currently cover this function

Reply:
Nope, I don't really know (yet) what is going on in `decorator.py` / `@validate_arguments`. All I know is that it uses `@validator` which is now deprecated, so I am replacing it with the new version.